### PR TITLE
Fixing flaky Template ID resolution test

### PR DESCRIPTION
**Pull request by leo-da**
Making sure that generated domain Template IDs do not have module entity duplicates, so the resolution should work with no problem.

``` shell
$ bazel test //ledger-service/http-json:tests --test_filter=com.daml.http.PackageServiceTest \
  --cache_test_results=no --runs_per_test=50
```
Passed 3 times. We should be all right now.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger-service/http-json/src/test/scala/com/digitalasset/http/Generators.scala [hunk 1]
```diff
@@ -26,17 +26,17 @@ object Generators {
       e <- Gen.identifier
     } yield domain.TemplateId(p, m, e)
 
-  def nonEmptySet[A](gen: Gen[A]): Gen[Set[A]] = Gen.nonEmptyListOf(gen).map(_.toSet)
+  def nonEmptySetOf[A](gen: Gen[A]): Gen[Set[A]] = Gen.nonEmptyListOf(gen).map(_.toSet)
 
   // Generate Identifiers with unique packageId values, but the same moduleName and entityName.
-  def genDuplicateApiIdentifiers: Gen[List[lav1.value.Identifier]] =
+  def genDuplicateModuleEntityApiIdentifiers: Gen[Set[lav1.value.Identifier]] =
```
Comment (cocreature) [inline ledger-service/http-json/src/test/scala/com/digitalasset/http/Generators.scala:32]: This still has `duplicate` in the name but it only returns uniques now. So what is duplicate?

Comment (S11001001) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/Generators.scala:32]: The `moduleName` and `entityName`.

Comment (cocreature) [reply ledger-service/http-json/src/test/scala/com/digitalasset/http/Generators.scala:32]: Oh I see, makes sense. I should read the comment :innocent:

**Review by cocreature — APPROVED** [review]
Thanks for the quick fix!

**Pull request merged**